pages/implementation/reference/kotlin.mdx

```diff
@@ -0,0 +1,208 @@
+    application: Application, context: Context
+)
+```
+
+Creates the Avo object that will be used to track. This method will initialize the SDKs for all destinations that are managed by Avo, and call the `make(env)` callback in all custom destinations.
+
+##### Arguments
+
+`env: AvoEnv`: Can be set to dev, prod and staging.
+
+`systemProperties`: a number of parameters equal to the number of system properties defined in your Avo workspace.
+The parameters are named the same as system properties, in camelCase, and require corresponding types: string, int, long, float, bool, list or any.
+
+`customDestination: ICustomDestination`: object, required if target source has a custom destination set up in Avo. Otherwise not present.
+
+Custom destination interface format:
+
+```kotlin
+interface ICustomDestination {
+    fun make(env: AvoEnv)
+
+    fun logEvent(eventName: String, eventProperties: Map<String, *>)
+
+    fun logPage(pageName: String, eventProperties: Map<String, *>) {}
+
+    fun setUserProperties(userId: String, userProperties: Map<String, *>)
+
+    fun identify(userId: String)
+
+    fun unidentify()
+}
+```
+
+Read more about custom destinations <Link passHref href="/implementation/start-custom-destination"><a>here</a></Link>.
+
+[//]: # 'TODO Find examples of Kotlin custom destinations for each analytics destination in <Link passHref href="/implementation/destinations"><a>this section</a></Link>.'
+
+`avoInspector: Any?`: optional Avo Inspector instance. If you use Avo Inspector pass it here to make the Avo functions automatically report the invocations to the Avo Inspector.
+
+`strict: Boolean`: optional bool, if set, Avo will throw an exception when it detects a tracking problem in development or staging. Note that the strict flag is ignored in production.
+
+###### Additional arguments
+
+`application: Application` and `context: Context` are required by some analytics destinations.
+
+#### setAvoLogger
+
+```kotlin
+avo.setAvoLogger { logMessage: String -> Log.d(this.localClassName, logMessage) }
+```
+
+This method allows you to provide custom implementation of the logger used by the Avo Functions.
+Can for example be used to disable logs or change which logging method is used.
+
+##### Arguments
+
+`avoLogger: (String) -> Unit`: custom implementation of the logger.
+
+#### setSystemProperties
+
+```kotlin
+avo.setSystemProperties(systemProprty0: Int, systemProperty1: Boolean?, ...)
+```
+
+A method to update system properties after initialization.
+
+##### Arguments
+
+`systemProperties`: a number of parameters equal to the number of system properties defined in your Avo workspace.
+The parameters are named the same as system properties, in camelCase, and require corresponding types: string, int, long, float, bool, list or any.
+
+#### Event tracking functions
+
+```kotlin
+avo.yourEventName(eventProperty0: Int, eventProperty1: Double?, ..., userProperty0: Boolean?, userProperty1: List<Int>, ..., userId_: String)
```

Comment:
How might we make it clear that `yourEventName` is not an actual function? Maybe something like:

```suggestion
avo.[yourEventName](eventProperty0: Int, eventProperty1: Double?, ..., userProperty0: Boolean?, userProperty1: List<Int>, ..., userId_: String)
```